pkg/data/domain/clientcert/service.go

```diff
@@ -60,13 +64,26 @@ func (s *Service) Delete(ctx context.Context, clientCert *ClientCert) error {
 	return nil
 }
 
-func (s *Service) GetCredential(ctx context.Context, namespace, name string) (*corev1.Secret, error) {
-	secret, err := s.client.K8sClient.K8sClient().CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return nil, microerror.Mask(notFoundError)
-	} else if err != nil {
+func (s *Service) GetCredential(ctx context.Context, clusterId, certUid string) (*corev1.Secret, error) {
+	selector := map[string]string{
+		kgslabel.Certificate: certUid,
+		apiextlabel.Cluster:  clusterId,
+	}
+
+	secrets, err := s.client.K8sClient.K8sClient().CoreV1().Secrets("").List(ctx, metav1.ListOptions{
```

Comment:
for consistency with other places in the code, could you use the controller client rather than the typed clientset?

Reply:
Are we using it elsewhere in here? I just used what was there before. But yeah let's change it

Reply:
On the same file I see we are using it

Reply:
Good will fix it thx

Reply:
done